crates/red_knot_python_semantic/src/types.rs

```diff
@@ -521,6 +532,26 @@ impl<'db> Type<'db> {
             {
                 true
             }
+            (Type::ClassLiteral(self_class), Type::SubclassOf(target_class)) => {
+                self_class.class.is_subclass_of(db, target_class.class)
+            }
+            (Type::SubclassOf(self_class), Type::SubclassOf(target_class)) => {
+                self_class.class.is_subclass_of(db, target_class.class)
+            }
+            (
+                Type::SubclassOf(SubclassOfType { class: self_class }),
+                Type::Instance(InstanceType {
+                    class: target_class,
+                    ..
+                }),
+            ) if self_class
+                .metaclass(db)
+                .into_class_literal()
+                .map(|meta| meta.class.is_subclass_of(db, target_class))
+                .unwrap_or(false) =>
+            {
+                true
```

Comment:
Looks good! My comment was vague because I hadn't thought through yet the handling of a non-class-literal metaclass here. But I think the only options here are class literal or dynamic type (Any/Unknown/Todo), and since this is `is_subtype_of`, ignoring dynamic type is correct.